skrub/_agg_joiner.py

```diff
@@ -263,17 +296,24 @@ def fit_transform(self, X, y=None):
         DataFrame
             The augmented input.
         """
-        self._check_inputs(X)
+        if isinstance(self.aux_table, str) and self.aux_table == "X":
+            self.aux_table = X
```

Comment:
we shouldn't modify `self.aux_table` but instead store a new fitted attribute `aux_table_` or private attribute `_aux_table`

Reply:
Do we need a copy since we're going to transform `_aux_table` ?